app/controllers/sessions_controller.rb

```diff
@@ -36,35 +36,30 @@ def auth_hash
   def authorize_access!
     return true unless Octobox.restricted_access_enabled?
 
-    client = Octokit::Client.new(access_token: auth_hash.credentials.token)
-    nickname = auth_hash.info.nickname
-
-    return true if organization_member?(client, nickname) || team_member?(client, nickname)
+    client = Octokit::Client.new(access_token: auth_hash.credentials.token, auto_paginate: true)
+    return true if organization_member?(client) || team_member?(client)
 
     flash[:error] = 'Access denied.'
     redirect_to root_path
   end
 
-  def organization_member?(*args)
-    member?(*args, type: :organization)
-  end
+  def organization_member?(client)
+    org_id = Octobox.config.github_organization_id
+    return false unless org_id
+
+    orgs = client.organizations(headers: { 'Cache-Control' => 'no-cache, no-store' })
```

Comment:
So going over this again, we should revert to using the https://developer.github.com/v3/orgs/members/#check-membership here.

It is scoped to the user making the request so even if they are part of the org privately they should still be able check their own membership.